client/dashboard/sites/scan/components/bulk-fix-threats-modal.tsx

```diff
@@ -27,69 +21,74 @@ export function BulkFixThreatsModal( { items, closeModal, siteId }: BulkFixThrea
 	const bulkFixableIds = new Set( bulkFixableThreats.map( ( item ) => item.id ) );
 	const remainingThreats = items.filter( ( item ) => ! bulkFixableIds.has( item.id ) );
 
-	const queryClient = useQueryClient();
-	const bulkFixThreats = useMutation( fixThreatsMutation( siteId ) );
 	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
-	const [ isBulkFixInProgress, setIsBulkFixInProgress ] = useState( false );
 
-	const { data: bulkFixStatusData } = useQuery( {
-		...fixThreatsStatusQuery( siteId, Array.from( bulkFixableIds ) ),
-		refetchInterval: isBulkFixInProgress ? 2000 : false,
-		enabled: isBulkFixInProgress,
-	} );
-
-	const handleBulkFixed = useCallback(
-		( message: string ) => {
-			queryClient.invalidateQueries( siteScanQuery( siteId ) );
-			queryClient.invalidateQueries( siteScanHistoryQuery( siteId ) );
-			closeModal?.();
-			createSuccessNotice( message, { type: 'snackbar' } );
-		},
-		[ closeModal, createSuccessNotice, queryClient, siteId ]
+	const { startFix, isFixing, status, error } = useFixThreats(
+		siteId,
+		Array.from( bulkFixableIds )
 	);
 
 	useEffect( () => {
-		if ( ! bulkFixStatusData?.threats ) {
-			return;
-		}
+		if ( status.isComplete && ! isFixing ) {
+			closeModal?.();
 
-		if ( isBulkFixInProgress ) {
-			const pendingThreats = bulkFixStatusData.threats.filter(
-				( threat ) => threat?.status === 'in_progress'
-			);
-			if ( pendingThreats.length > 0 ) {
-				return;
+			if ( status.allFixed ) {
+				createSuccessNotice(
+					_n( 'Threat fixed.', 'All threats were successfully fixed.', bulkFixableThreats.length ),
```

Comment:
Added pluralization because there could be a case where only 1 threat is available for bulk fixing:

<img width="60%" alt="CleanShot 2025-09-28 at 14 26 15@2x" src="https://github.com/user-attachments/assets/976df7f8-4fd0-46ef-9b91-1118bb5a7c58" />